multinet/db.py

```diff
@@ -181,12 +200,15 @@ def get_workspace(name: str) -> WorkspaceSpec:
 
 # Caches the reference to the StandardDatabase instance for each workspace
 @lru_cache()
-def get_workspace_db(name: str) -> StandardDatabase:
+def get_workspace_db(name: str, readonly: bool = False) -> StandardDatabase:
     """Return the Arango database associated with a workspace, if it exists."""
     doc = workspace_mapping(name)
     if not doc:
         raise WorkspaceNotFound(name)
 
+    if readonly:
+        return read_only_db(doc["internal"])
+
     return db(doc["internal"])
```

Comment:
The following is more pythonic:

```suggestion
    name = doc["internal"]
    return read_only_db(name) if readonly else db(name)
```